### PR TITLE
Prepare loading GHMHD initial data from files

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Top is the ID grid and bottom is the evolution grid:

Bildschirm­foto 2023-03-15 um 15 36 10

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/GrMhd/GhValenciaDivClean/BoundaryConditions/Factory.hpp [hunk 1]
```diff
@@ -62,7 +62,6 @@ using standard_boundary_conditions = tmpl::push_back<
 
 /// Boundary conditions that work with finite difference.
 using standard_fd_boundary_conditions =
-    tmpl::list<ConstraintPreservingFreeOutflow, DirichletAnalytic,
```
Comment (nilsdeppe) [inline src/Evolution/Systems/GrMhd/GhValenciaDivClean/BoundaryConditions/Factory.hpp:65]: Why do we need to change the definition of standard? Please keep as-is

File: src/PointwiseFunctions/Hydro/Tags.hpp [hunk 2]
```diff
@@ -16,6 +16,19 @@
 /// \ingroup EvolutionSystemsGroup
 /// \brief Items related to hydrodynamic systems.
 namespace hydro {
+
+/// %Tags for options of hydrodynamic systems.
+namespace OptionTags {
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/Hydro/Tags.hpp:21]: Move these changes into a separate commit

File: src/Evolution/Executables/GrMhd/GhValenciaDivClean/GhValenciaDivCleanBase.hpp [hunk 3]
```diff
@@ -318,10 +318,25 @@ template <bool UseDgSubcell,
...
     EvolutionMetavarsDerived<InitialData, InterpolationTargetTags...>,
-    UseDgSubcell> : public virtual GhValenciaDivCleanDefaults<UseDgSubcell> {
+    UseDgSubcell>
+    : public virtual GhValenciaDivCleanDefaults<
+          UseDgSubcell, evolution::is_numeric_initial_data_v<InitialData>> {
   using derived_metavars =
       EvolutionMetavarsDerived<InitialData, InterpolationTargetTags...>;
-  using defaults = GhValenciaDivCleanDefaults<UseDgSubcell>;
+
+  static constexpr bool use_dg_subcell = UseDgSubcell;
```
Comment (nilsdeppe) [inline src/Evolution/Executables/GrMhd/GhValenciaDivClean/GhValenciaDivCleanBase.hpp]: I don't understand why any of the changes in this commit are happening. This does more than the template parameter. Also, in a couple commits I ask if we can check locally if the analytic tag is retrievable, and if not then we assume numeric initial data. I think if that's possible then we can skip this commit completely.

File: src/Evolution/DgSubcell/Actions/Initialize.hpp [hunk 4]
```diff
@@ -198,17 +204,33 @@ struct Initialize {
         },
         make_not_null(&box));
     if (cell_is_troubled) {
-      // Set variables on subcells.
-      if constexpr (System::has_primitive_and_conservative_vars) {
-        db::mutate<typename System::primitive_variables_tag>(
-            make_not_null(&box), [&subcell_mesh](const auto prim_vars_ptr) {
-              prim_vars_ptr->initialize(subcell_mesh.number_of_grid_points());
-            });
+      if constexpr (UseNumericInitialData) {
```
Comment (nilsdeppe) [inline src/Evolution/DgSubcell/Actions/Initialize.hpp]: Can you more easily check if the `AnalyticData` or `AnalyticSolution` tag is in the box? Then we don't need to add a template parameter to this action. I honestly don't love how invasive the numerical initial data is currently (not just in this PR). It causes an avalanche of template parameters to propagate through a lot of the code. It's almost as invasive as DG-FD, I find.

File: src/Evolution/Executables/GrMhd/GhValenciaDivClean/GhValenciaDivCleanBase.hpp
Comment (nilsdeppe) [inline src/Evolution/Executables/GrMhd/GhValenciaDivClean/GhValenciaDivCleanBase.hpp:1]: Can you specify in the commit message why this needs to be done, please?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
I'd like to deal with the last 2 commits in a separate PR. This is a rather hefty PR as-is

File: src/Evolution/Systems/GrMhd/GhValenciaDivClean/BoundaryConditions/Factory.hpp (hunk 1 above)
Comment (nilsvu) [reply src/Evolution/Systems/GrMhd/GhValenciaDivClean/BoundaryConditions/Factory.hpp:65]: As mentioned earlier the DirichletAnalytic boundary condition doesn't work with numeric initial data. Since the `standard_fd_boundary_conditions` alias is only used in the executable and in tests I removed DirichletAnalytic here and appended it to the list where it's needed.

File: src/Evolution/Executables/GrMhd/GhValenciaDivClean/GhValenciaDivCleanBase.hpp (hunk 3 above)
Comment (nilsvu) [reply src/Evolution/Executables/GrMhd/GhValenciaDivClean/GhValenciaDivCleanBase.hpp]: dropped this commit

File: src/Evolution/DgSubcell/Actions/Initialize.hpp (hunk 4 above)
Comment (nilsvu) [reply src/Evolution/DgSubcell/Actions/Initialize.hpp]: done

File: src/Evolution/Executables/GrMhd/GhValenciaDivClean/GhValenciaDivCleanBase.hpp
Comment (nilsvu) [reply src/Evolution/Executables/GrMhd/GhValenciaDivClean/GhValenciaDivCleanBase.hpp:1]: done

**Comment by nilsvu** [comment]
Made updates and shortened the PR.

**Review by nilsdeppe — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
CI is very unhappy

**Review by nilsdeppe — APPROVED** [review]
